website/server/controllers/api-v3/user.js

```diff
@@ -191,17 +191,22 @@ api.deleteUser = {
     let user = res.locals.user;
     let plan = user.purchased.plan;
 
-    req.checkBody({
-      password: {
-        notEmpty: {errorMessage: res.t('missingPassword')},
-      },
-    });
-
     let validationErrors = req.validationErrors();
     if (validationErrors) throw validationErrors;
 
-    let oldPassword = passwordUtils.encrypt(req.body.password, user.auth.local.salt);
-    if (oldPassword !== user.auth.local.hashed_password) throw new NotAuthorized(res.t('wrongPassword'));
+    if(user.auth.local.hashed_password && user.auth.local.email) {
+      req.checkBody({
```

Comment:
req.checkBody doesn't do anything unless you call req.validationErrors() and then throw the error if it exists, like it's done a few lines above